src/base.jl

```diff
@@ -11,3 +11,23 @@ function spin_hamming_distance(spin_1, spin_2)
     end
     return hamming_dist
 end
+
+function _get_states(ising_model, energy_levels; n = 0)
+    states = nothing
+    if (ising_model == nothing && n == 0)
+        error("both ising model and n were not specified, could not compute states")
+    elseif ising_model == nothing && energy_levels != 0
+        error("must provide ising model if energy_levels != 0")
+    elseif ising_model != nothing && energy_levels != 0
+        energies = _QA.compute_ising_energy_levels(ising_model)
+        energy_levels = (energy_levels == 0) ? length(energies) : energy_levels
```

Comment:
This condition `(energy_levels == 0)` will never be true at this point in code as enforced by `elseif ising_model != nothing && energy_levels != 0`.  

However, I do belive you need something of the form `energy_levels = min(energy_levels, length(energies))` otherwise if the user specifies too many then `[energies[i].states for i = 1:energy_levels]` will get an index out of range.

You might want to add a test for this situation. :-)